### PR TITLE
8342183: Update tests to use stronger algorithms and keys

### PR DESCRIPTION
**Pull request by fguallini**
Several tests are identified to use weak/legacy algorithms. As these tests purpose is not to exercise weak algorithms, these are updated in this PR to use stronger algorithms. If the test uses a weak key, it is updated as well.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8342183](https://bugs.openjdk.org/browse/JDK-8342183): Update tests to use stronger algorithms and keys (**Sub-task** - P2)

### Reviewers
 * [Rajan Halade](https://openjdk.org/census#rhalade) (@rhalade - **Reviewer**)
 * [Anthony Scarpino](https://openjdk.org/census#ascarpino) (@ascarpino - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21578/head:pull/21578` \
`$ git checkout pull/21578`

Update a local copy of the PR: \
`$ git checkout pull/21578` \
`$ git pull https://git.openjdk.org/jdk.git pull/21578/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21578`

View PR using the GUI difftool: \
`$ git pr show -t 21578`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21578.diff">https://git.openjdk.org/jdk/pull/21578.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21578#issuecomment-2443747675)
</details>

### CODE REVIEW

File: test/jdk/com/sun/crypto/provider/KeyAgreement/SupportedDHParamGens.java
Comment (mpdonova) [inline test/jdk/com/sun/crypto/provider/KeyAgreement/SupportedDHParamGens.java:61]: It looks like the DiffieHelmanGroup fields are just used to intialize the DHParameterSpec. It might be cleaner to create the DHParameterSpec objects in a static initializer. 

Also, the DHParameterSpec objects can be `final`

Comment (fguallini) [reply test/jdk/com/sun/crypto/provider/KeyAgreement/SupportedDHParamGens.java]: Actually, these vars are not needed, simplified with latest commit.

**Comment by rhalade** [comment]
The changes in this PR should not be integrated until https://github.com/openjdk/jdk/pull/21537 is integrated. The updates in SecurityUtils should be used from https://github.com/openjdk/jdk/pull/21537.

File: test/jdk/com/sun/crypto/provider/KeyAgreement/SupportedDHParamGensLongKey.java
Comment (ascarpino) [inline test/jdk/com/sun/crypto/provider/KeyAgreement/SupportedDHParamGensLongKey.java:27]: It looks like you can remove this change & file from the PR

File: test/jdk/java/security/Signature/ByteBuffers.java
Comment (ascarpino) [inline test/jdk/java/security/Signature/ByteBuffers.java:53]: Why is SHA224 used instead of SHA256 here and on other tests in this PR?

Comment (fguallini) [reply test/jdk/java/security/Signature/ByteBuffers.java]: I will update to use SHA256 instead, thanks

File: test/jdk/com/sun/crypto/provider/KeyAgreement/SupportedDHParamGensLongKey.java
Comment (fguallini) [reply test/jdk/com/sun/crypto/provider/KeyAgreement/SupportedDHParamGensLongKey.java:27]: The reason for this, is that this test is calling another : SupportedDHParamGens which uses the SecurityUtils, it does need to build /test/lib to compile

**Comment by rhalade** [comment]
Thanks for updating PR with suggested changes. The updates look good to me. I recommend to wait for another reviewer (@ascarpino ?) to approve as well.

**Review by rhalade — APPROVED** [review]

**Review by ascarpino — APPROVED** [review]

**Pull request closed without merging**
